src/worker.h

```diff
@@ -0,0 +1,25 @@
+#ifndef WORKER_H
+#define WORKER_H
+
+#include "nan.h"
+#include "spellchecker.h"
+
+#include <vector>
+
+using namespace spellchecker;
+using namespace v8;
+
+class CheckSpellingWorker : public Nan::AsyncWorker {
+public:
+  CheckSpellingWorker(const std::vector<uint16_t> &&corpus, SpellcheckerImplementation* impl, Nan::Callback* callback);
```

Comment:
I think that `corpus` shouldn't be `const`, because the constructor actually *does* mutate that parameter (by moving its contents elsewhere). In general, you don't normally use `const` with rvalue reference parameters for that reason.

I guess it actually works anyway because the `std::move` in the implementation casts away the `const`-ness, but removing the `const` would document the constructor's behavior more truthfully.